CMakeLists.txt

```diff
@@ -1,4 +1,4 @@
-cmake_minimum_required (VERSION 2.8)
+cmake_minimum_required (VERSION 2.8.12)
```

Comment:
looks like travis is stuck at 2.8.7. We're using that as a proxy for minspec as it is Ubuntu 12.04 LTS which seems like a good base system.

having said that, we could add something like https://gist.github.com/winterz/10cc2741d466cbd8ff21 to the travis build script to get the Trusty cmake (which is 2.8.12 iiuc).